store/tikv/2pc.go

```diff
@@ -920,6 +920,9 @@ func (c *twoPhaseCommitter) execute(ctx context.Context) (err error) {
 		if c.isOnePC() {
 			// The error means the 1PC transaction failed.
 			if err != nil {
+				if c.getUndeterminedErr() == nil {
+					c.cleanup(ctx)
```

Comment:
Should we just use something like `pessimisticRollbackMutations` if it's pessimistic transactions instead of `cleanup`.

Reply:
We can always use pessimistic rollback because a optimistic one pc won't leave any waste.

Reply:
How about changing `cleanup()`?
```golang
		cleanupKeysCtx := context.WithValue(context.Background(), TxnStartKey, ctx.Value(TxnStartKey))
		var err error
		if c.isPessimistic && c.isOnePC() {
			err = c.pessimisticRollbackMutations(NewBackofferWithVars(cleanupKeysCtx, cleanupMaxBackoff, c.txn.vars), c.mutations)
		} else {
			err = c.cleanupMutations(NewBackofferWithVars(cleanupKeysCtx, cleanupMaxBackoff, c.txn.vars), c.mutations)

		}
		if err != nil {
```

Reply:
@youjiali1995 We needn't do cleanup for optimistic 1PC, so I suggest this:

```go
if !c.isOnePC() {
	err = c.cleanupMutations(NewBackofferWithVars(cleanupKeysCtx, cleanupMaxBackoff, c.txn.vars), c.mutations)
} else if c.isPessimistic {
	err = c.pessimisticRollbackMutations(NewBackofferWithVars(cleanupKeysCtx, cleanupMaxBackoff, c.txn.vars), c.mutations)
}
```

Reply:
My mistake.